### PR TITLE
Implementing viewing functionality for projects profiles #13

### PR DESCRIPTION
**Pull request by Thoughtseize1**
Implemented permission checks for each method in ProjectViewSet.

**Created two custom methods:**
- invest_to_project: Allows investors to participate in projects. URL: `/api/projects/3/invest/`
The method currently checks access permissions and verifies whether the user has already invested in the project.
- get_my_projects: Enables investors to view their own projects. URL: `/api/projects/my`

**Implemented project viewing capabilities:**
`/api/projects`: Users can view limited information about all projects.
`/api/projects/12`: Users can view a specific project by ID.

- Investor users can see full project information (including budget and contact details) thanks to the serializer.
- Investor users can view projects they are involved in (by investing through the invest_to_project method).
- Implemented filtering by query parameters:
`/api/projects?project_name=la_&bgt=5000`
`/api/projects?project_name=la_&bgt=5000&blt=9000`
- Filtering is implemented based on query parameters. Investor users can filter projects based on the budget_needed field, while non-investor users cannot use budget filtering.
- Implemented filtering projects by a part of the project name:
`/api/projects?project_name=la_`
- The following fields are allowed for filtering: `'project_name', 'description', 'industry', 'status', 'bgt', and 'blt'`.
```
bgt - budget greater or than 
blt - budget less or equal than 
```

### CODE REVIEW

File: projects/views.py [hunk 1]
```diff
@@ -152,3 +163,51 @@ def partial_update(self, request, pk):
...
+        :return: Response: A response with a detailed message about the result of the operation.
+        Examples:
+        Use this method by sending a POST request to the URL: /api/projects/12/invest
+        where 12 is the ID of the project you want to add an investor to.
+        """
+
+        try:
+            user = request.user
+            project = Project.objects.get(id=pk, is_active=True)
+            investor = Investor.objects.get(user=user, is_active=True)
```
Comment (superolegatron) [inline projects/views.py:193]: maybe it is worth using get_object_or_404

**Review by superolegatron — APPROVED** [review]
Seems ok

File: projects/views.py (hunk 1 above)
Comment (Thoughtseize1) [reply projects/views.py:220]: Added ✅

File: projects/views.py [hunk 2]
```diff
@@ -216,8 +215,8 @@ def invest_to_project(self, request, pk=None):
 
         try:
             user = request.user
-            project = Project.objects.get(id=pk, is_active=True)
-            investor = Investor.objects.get(user=user, is_active=True)
+            project = get_object_or_404(Project, is_active=True, id=pk)
+            investor = get_object_or_404(Investor, user=user, is_active=True)
```
[the remark this answers is not in the document]
Comment (Thoughtseize1) [reply projects/views.py]: Done ✅

**Comment by OlesiaBarannik** [comment]
All methods work well

**Review by OlesiaBarannik — APPROVED** [review]

**Review by VovaD199 — APPROVED** [review]

**Pull request merged**